js/ssai_pulse.js

```diff
@@ -209,13 +230,28 @@ OO.Ads.manager(function(_, $)
      * @param {string} eventname The name of the event for which this callback is called
      * @param {number} playhead The total amount main video playback time (seconds)
      * @param {number} duration Duration of the live video (seconds)
-     * @param {number} livePlayhead The current playhead within the DVR/live window (seconds)
+     * @param {number} offset Current video time (seconds). Currently is obtained just for live stream from amc.
      */
-    this.onPlayheadTimeChanged = function(eventName, playhead, duration, livePlayhead) {
-      var offset = duration - livePlayhead;
-      if (_.isFinite(offset) && offset >= 0)
+    
+    this.onPlayheadTimeChanged = function(eventName, playhead, duration, offset) {
+      var offsetParam = 0;
+      
+      if (!amc.isLiveStream) 
       {
-        currentOffset = offset;
+        if (duration && typeof duration === 'number' && duration > 0)
+        {
+          offsetParam = duration - playhead;
+        }
+      }
+      //For live streams, if user moved the playback head into the past, offset is the seconds in the past that user is watching
+      if ((amc.isLiveStream && (offset && typeof offset === 'number') && (duration && typeof duration === 'number')) && offset > 0 && offset < duration) 
```

Comment:
@cacel underscore has a `_.isNumber` function that we use to determine if a value is numeric. You can see some examples in ad_manager_vast.js. Please use that instead of `typeof`